internal/onboarding/azure_test.go

```diff
@@ -192,41 +189,150 @@ resource "time_sleep" "wait_time" {
 }
 
+resource "azurerm_role_assignment" "mondoo_security-0" {
+  principal_id       = azuread_service_principal.mondoo.object_id
+  role_definition_id = azurerm_role_definition.mondoo_security.role_definition_resource_id
+  scope              = "/subscriptions/abc-123-xyz-456-1"
+}
+
 resource "azurerm_role_assignment" "reader-1" {
   principal_id         = azuread_service_principal.mondoo.object_id
   role_definition_name = "Reader"
   scope                = "/subscriptions/abc-123-xyz-456-2"
 }
 
+resource "azurerm_role_assignment" "mondoo_security-1" {
+  principal_id       = azuread_service_principal.mondoo.object_id
+  role_definition_id = azurerm_role_definition.mondoo_security.role_definition_resource_id
+  scope              = "/subscriptions/abc-123-xyz-456-2"
+}
+
 resource "azurerm_role_assignment" "reader-2" {
   principal_id         = azuread_service_principal.mondoo.object_id
   role_definition_name = "Reader"
   scope                = "/subscriptions/abc-123-xyz-456-3"
 }
 
+resource "azurerm_role_assignment" "mondoo_security-2" {
+  principal_id       = azuread_service_principal.mondoo.object_id
+  role_definition_id = azurerm_role_definition.mondoo_security.role_definition_resource_id
+  scope              = "/subscriptions/abc-123-xyz-456-3"
+}
+
 resource "mondoo_integration_azure" "this" {
   client_id = azuread_application.mondoo.client_id
   credentials = {
     pem_file = join("\n", [tls_self_signed_cert.credential.cert_pem, tls_private_key.credential.private_key_pem])
   }
-  depends_on              = [azuread_service_principal.mondoo, azuread_application_certificate.mondoo, azuread_directory_role_assignment.readers, azurerm_role_assignment.reader-0, azurerm_role_assignment.reader-1, azurerm_role_assignment.reader-2]
+  depends_on              = [azuread_service_principal.mondoo, azuread_application_certificate.mondoo, azuread_directory_role_assignment.readers, azurerm_role_definition.mondoo_security, azurerm_role_assignment.reader-0, azurerm_role_assignment.mondoo_security-0, azurerm_role_assignment.reader-1, azurerm_role_assignment.mondoo_security-1, azurerm_role_assignment.reader-2, azurerm_role_assignment.mondoo_security-2]
   name                    = "test-integration"
   scan_vms                = true
   subscription_allow_list = ["abc-123-xyz-456-1", "abc-123-xyz-456-2", "abc-123-xyz-456-3"]
   tenant_id               = data.azuread_client_config.current.tenant_id
 }
+`
+	assert.Equal(t, expected, code)
+}
 
-resource "azurerm_role_assignment" "mondoo_security" {
-  count              = length(data.azurerm_subscriptions.available.subscriptions)
-  principal_id       = azuread_service_principal.mondoo.object_id
-  role_definition_id = azurerm_role_definition.mondoo_security.role_definition_resource_id
-  scope              = data.azurerm_subscriptions.available.subscriptions[count.index].id
+func TestGenerateAzureHCLScanVMsForAllSubscriptions(t *testing.T) {
```

Comment:
Would be nice to also have a test case like "TestGenerateAzureHCLScanVMsForAllowedSubscriptions" with `integration.Allow` being set.